### ISSUE TITLE
Opening a panel element makes the panel go opaque at all cases

### ISSUE THREAD

**Issue by Tromzy** [psifidotos/Latte-Dock#956]
On Latte Dock 0.7.79, when I open a Plasma panel element (K menu, systray, calendar, etc.), the panel goes from transparent to opaque. I tried to play with every setting (including dynamic backgrounds) put I was not able to disable it.

It would be nice to have an option to restore previous behavior (panel staying transparent / blurred when opening a panel element).

**Comment by psifidotos** [comment]
Οκ... I wanted you to report this properly because this is a design decision not an option to just add a choice and as such needs a detailed description how everything is going to interact which each other. The result of the discussion is going to provide the criteria under which in the future something will be considered a bug or design decision-wont fix case.

Latte provides the following options concerning panel background transparency and visibility:

**A:** Appearance Page -> Show Background
**B:** Appearance Page -> Transparency slider
**C:** Appearance Page -> Solid Panel
**D:** Tweaks Page -> Force Solid Background for maximized/snapped windows
**D->V:** Improve contents visibility when panel is transparent
**E:** Tweaks Page ->  Hide Background for not maximized windows
**F:** Automatic identify expanded applets and make panel transparent
     - _(based on the following cases this should be activated only when D is enabled)_

**Current Design Decision:**
_Latte provides only TWO transparency levels for panel background one of which should always be fully transparent or fully solid. Based on that the user cant have two transparency levels that are in between levels, such as: 20% and 75%_

### Options Combinations

**Case 1: [A] is Disabled**
1. The panel is transparent at all cases except:
a. **[D]** is enabled and there is a maximized/snapped window
2. **[F]** isnt applied  at any case because **[A]** is disabled

**Case 2: [B] has mid value(not 0% or 100%) and neither [D] or [E] are enabled**
1. The panel is always at the **[B]**-set transparency level
2. **[F]** isnt applied except if the user has enabled **[D]**

**Case 3:[B] has mid value(not 0% or 100%),  [D] is enabled and [E] is not enabled**
1. The panel is always at the **[B]**-set transparency level except when there is a maximized or snapped window because of **[D]**
2. **[F]** should be applied for expanded applets because **[D]** is enabled

**Case 4:[B] has mid value(not 0% or 100%),  [D] is not enabled and [E] is enabled**
1. The panel is always transparent because of **[E]** option and when there is a maximized or snapped window it should change to **[B]**-set transparency level
2. **[F]** wont be applied for expanded applets because **[D]** is disabled

**Case 5:[C] is Enabled**

C option provides a solid background from the plasma theme. So for [C] option Case 2,3,4 are applied like the [B] value is 100%


----

### Defaults

Based on the above design decisions and explanations the most-used case should be the default and that is: **[A] enabled and [F] enabled** . For **[A]** that was already the case. The importance of **[F]** is that it provides always a plasma theme related feeling.

**Comment by Tromzy** [comment]
Sorry, I don't understand your answer at all. What am I supposed to answer ? I only want the behaviour that was on the previous Latte Dock versions, that is : do not make the panel opaque when I open something on it.

**Comment by psifidotos** [comment]
this issue is related to many other options so before providing it I wanted a documentation how everything should work with each other. There are more users that reported that issue but most of them may have a different understanding how Dynamic Background options and Panel Backgrounds should interact with each other. The above documentation will be the guide for bugs reporters and developers. According to the above documentation what you are describing is a bug.

**_If you havent enabled [D] then in such case expanding applets shouldnt make the panel solid._**

**Comment by Tromzy** [comment]
Thank you, it's clearer now. I did not enable [D] so I think I am encounting a bug indeed. What can I provide to you to fix it ?

**Comment by psifidotos** [comment]
Based on #957 this is a bug... The discussion concerning the Design Decision moved to the previous thread.

@Tromzy nothing more is needed from you :) thanks a lot for reporting... I just need to implement #957 and every single detail in there to be taken into account... I think it will take just a few days.

**Comment by Tromzy** [comment]
Yes, it works ! Thanks a lot for being so fast !
